### PR TITLE
Add mediatype to datapoint

### PR DESCRIPTION
**Pull request by Ljinod**
This step is the first in enabling the sharing of files between Cozy
instances: we need a reliable way - on the server side - to extract the
url of a Cozy cloud from a contact.

The solution we chose was to add a mediatype to the datapoint structure:
we can interpret an url according to the value of its mediatype. For
instance if the mediatype is 'cloud:cozy' we know that the url
represents the address of a Cozy instance and if the mediatype is empty
it's simply an email address.

This solution has the advantages of still being compliant to the vCard
RFC and, normally, being transparent.

### CODE REVIEW

File: client/app/views/contacts/components/datapoint.coffee [hunk 1]
```diff
@@ -23,11 +23,30 @@ module.exports = class ContactDatapointView extends Mn.ItemView
...
+        # elements selected from the "other information" dropdown list.
         entity = switch data.name
             when 'social', 'chat' then 'social'
-            when 'url'            then 'url'
+
+            # So that we are still compliant to the vCard RFC we need to
+            # convert the data.name from 'share' to 'url' when we store it.
+            # Once this is done we still need to set their entity to 'share'.
+            when 'url'
+                if data.mediatype? and (data.mediatype.search 'cloud' != -1)
```
Comment (m4dz) [inline client/app/views/contacts/components/datapoint.coffee]: Parenthesis are useless here, you can safely remove them

**Comment by m4dz** [comment]
Thanks for you contribution @Ljinod!

It seems to be ok for me, can you just remove your `build` form your commits, as long as we don't commit builds into the _development_ branch anymore? Thanks!

**Comment by Ljinod** [comment]
Thanks for your feedback @m4dz!

I've removed the `build` from my commits as well as `helpers.jade` - were I had just added two blank lines.

**Comment by m4dz** [comment]
Thanks a lot :+1:!

**Pull request merged**